File: spring_tofu_app/src/main/java/com/trianaSalesianos/tofuApp/repository/UserRepository.java
package com.trianaSalesianos.tofuApp.repository;

import com.trianaSalesianos.tofuApp.model.Recipe;
import com.trianaSalesianos.tofuApp.model.User;
import com.trianaSalesianos.tofuApp.model.dto.user.UserResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import javax.transaction.Transactional;
import java.util.Collection;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface UserRepository extends JpaRepository<User, UUID>, JpaSpecificationExecutor<User> {

    Optional<User> findFirstByUsername(String username);
    boolean existsByUsername(String username);

}

File: mobile_tofu_app/app/src/main/java/com/example/tofuapp/data/repository/UserRepository.kt
package com.example.tofuapp.data.repository

import com.example.tofuapp.data.service.UserService
import com.example.tofuapp.util.apiRequestFlow
import javax.inject.Inject

class UserRepository @Inject constructor(
   private val userService: UserService,
) {
    fun getCurrentUser() = apiRequestFlow {
        userService.getCurrentUser()
    }
}